edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/handlers/ScopeIdentitiesHandler.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Hub.Core;
+    using Microsoft.Azure.Devices.Edge.Hub.Core.Identity.Service;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Edge.Util.Concurrency;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    /// <summary>
+    /// ScopeIdentitiesHandler is responsible for syncing authorized identities from edgeHub core to the Mqtt Broker.
+    /// </summary>
+    public class ScopeIdentitiesHandler : IMessageProducer
+    {
+        const string Topic = "$internal/identities";
+        readonly IDeviceScopeIdentitiesCache deviceScopeIdentitiesCache;
+        readonly AtomicBoolean connected = new AtomicBoolean(false);
+        IMqttBrokerConnector connector;
+        IList<BrokerServiceIdentity> lastBrokerServiceIdentityUpdate = new List<BrokerServiceIdentity>();
+
+        public ScopeIdentitiesHandler(IDeviceScopeIdentitiesCache deviceScopeIdentitiesCache)
+        {
+            this.deviceScopeIdentitiesCache = deviceScopeIdentitiesCache;
+            this.deviceScopeIdentitiesCache.ServiceIdentitiesUpdated +=
+                async (sender, serviceIdentities) => await this.ServiceIdentitiesUpdated(serviceIdentities);
+        }
+
+        public void SetConnector(IMqttBrokerConnector connector)
+        {
+            this.connector = connector;
+            this.connector.OnConnected += async (sender, args) => await this.OnConnect();
+        }
+
+        async Task OnConnect()
+        {
+            this.connected.Set(true);
+            await this.PublishBrokerServiceIdentities(this.lastBrokerServiceIdentityUpdate);
```

Comment:
>this.lastBrokerServiceIdentityUpdate [](start = 54, length = 36)

I don't see anything guaranteeing this is populated at OnConnect time.  Is it going for the list to be empty if ServiceIdentitiesUpdated fires later than OnConnect?

Reply:
There's a check at the beginning of PublishBrokerServiceIdentities that will do a no-op if the list is empty